dashboard/src/pages/TreeDetails/TreeDetailsFilter.tsx

```diff
@@ -0,0 +1,260 @@
+import {
+  Dispatch,
+  memo,
+  SetStateAction,
+  useCallback,
+  useMemo,
+  useState,
+} from 'react';
+import { FormattedMessage, useIntl } from 'react-intl';
+
+import { z } from 'zod';
+
+import { useNavigate, useParams } from '@tanstack/react-router';
+
+import FilterDrawer from '@/components/Filter/Drawer';
+import FilterSummarySection from '@/components/Filter/SummarySection';
+import FilterCheckboxSection, {
+  ICheckboxSection,
+} from '@/components/Filter/CheckboxSection';
+import {
+  TFilter,
+  TFilterKeys,
+  TreeDetails as TreeDetailsType,
+} from '@/types/tree/TreeDetails';
+import { useTreeDetails } from '@/api/TreeDetails';
+import { Skeleton } from '@/components/ui/skeleton';
+
+import { cleanFalseFilters } from './treeDetailsUtils';
+
+const zFilterValue = z.record(z.boolean());
+type TFilterValues = z.infer<typeof zFilterValue>;
+
+interface ITreeDetailsFilter {
+  filter: TFilter;
+  treeUrl: string;
+}
+
+const filterFieldMap = {
+  git_repository_branch: 'branches',
+  config_name: 'configs',
+  architecture: 'archs',
+  valid: 'status',
+} as const satisfies Record<string, TFilterKeys>;
+
+export const mapFilterToReq = (
+  filter: TFilter,
+): { [key: string]: string[] } => {
+  const filterMapped: { [key: string]: string[] } = {};
+
+  Object.entries(filterFieldMap).forEach(([reqField, field]) => {
+    const values = filter[field];
+    if (!values) return;
+
+    Object.entries(values).forEach(([value, isSelected]) => {
+      if (isSelected) {
+        if (reqField == 'valid') {
+          value = value == 'Sucess' ? 'true' : 'false';
+        }
+        if (!filterMapped[reqField]) filterMapped[reqField] = [];
+        filterMapped[reqField].push(value);
+      }
+    });
+  });
+
+  return filterMapped;
+};
+
+export const createFilter = (data: TreeDetailsType | undefined): TFilter => {
+  const status = { Sucess: false, Failure: false };
```

Comment:
typo: success

Reply:
These files were moved, I didn't change here because I was afraid of collateral